test/unittests/test_skill_manager.py

```
@@ -358,3 +409,180 @@ def test_load_plugin_skill_returns_false(self):
+        'skills': {**mock_config()['skills'], 'use_deferred_loading': True}
+    })
+    def test_deferred_loading_enabled_via_config(self):
+        """Test that deferred loading can be enabled via config."""
+        skill_manager = SkillManager(self.message_bus_mock)
+
+        self.assertTrue(skill_manager._use_deferred_loading)
+
+    @patch.dict(Configuration._Configuration__patch, mock_config())
+    def test_connectivity_handlers_not_registered_when_deferred_loading_disabled(self):
+        """Test that connectivity event handlers are NOT registered when deferred loading is disabled."""
+        skill_manager = SkillManager(self.message_bus_mock)
```

Ruff (0.15.5)

[error] Local variable `skill_manager` is assigned to but never used

Remove assignment to unused variable `skill_manager`

(F841)